codex/logutils.nim

```diff
@@ -184,7 +184,7 @@ template formatIt*(format: LogFormat, T: typedesc, body: untyped) =
       let v = opts.map(opt => opt.formatJsonOption)
       setProperty(r, key, json.`%`(v))
 
-    proc setProperty*(r: var JsonRecord, key: string, val: seq[T]) =
+    proc setProperty*(r: var JsonRecord, key: string, val: seq[T]) {.upraises:[ValueError, IOError].} =
```

Comment:
I want someone to tell me whether this should be applied to all of these procs. Is it going to cause people confusion if at some point in the future they use this without knowing it and it'll crash compilation with an error that's unrelated to what they are doing?

Reply:
There's going to be a separate PR to fix the logutils raises issue. Then we'll rebase this, then we can review

Reply:
You can start adding the raises clauses IMHO. And it's `raises` not `upraises`.

Reply:
Should I apply this same change to the other setProperty procs in the same file that already have exactly the same thing?